testStrategies/suiteBasedTesting.go

```diff
@@ -102,29 +101,67 @@ func executeTestSuite(
 
 			targetHost, targetPort := determineHostandPortforRequest(testDefinition, configurationSettings)
 			responseTime := testDefinition.BuildAndSendRequest(configurationSettings.RequestDelay, targetHost, targetPort, uniqueTestRunID, globalsMap)
```

Comment:
The code from this point to line 158 increment the error counters across all threads as atomic values. The nested conditionals on lines 135-139 and again from 145-158 are to ensure the variables have been created in the data structure before attempting to increment.